File: src/main/scala/booklet/pure/http/CustomResponse.scala
package booklet.pure.http

import booklet.pure.Failure
import zhttp.http.Status.SeeOther
import zhttp.http._

object CustomResponse {

  def ok(body: Body, contentType: String): Response =
    Response(
      headers = Headers("content-type" -> contentType),
      body = body
    )

  def ok(body: Body): Response =
    ok(body, contentType = "text/html")

  def okJs(body: Body): Response =
    ok(body, contentType = "text/javascript")

  def seeOther(path: String): Response =
    Response(
      status = SeeOther,
      headers = Headers("location" -> path)
    )

  def badRequest(message: String): Response =
    Response.fromHttpError(HttpError.BadRequest(message))

  def notFound(path: Path): Response =
    Response.fromHttpError(HttpError.NotFound(path.toString))

  def serverFailure(failure: Failure): Response =
    Response.fromHttpError(HttpError.InternalServerError(failure.message, failure.cause))
}

File: src/main/ts/booklet/pure/http/CustomResponse.ts
import { Response, Headers, Body, HttpError, Status } from 'some-http-library';
import { Failure } from '../Failure';

export class CustomResponse {
  static ok(body: Body, contentType: string = 'text/html'): Response {
    return new Response({
      headers: new Headers({ 'content-type': contentType }),
      body: body
    });
  }

  static okJs(body: Body): Response {
    return this.ok(body, 'text/javascript');
  }

  static seeOther(path: string): Response {
    return new Response({
      status: Status.SeeOther,
      headers: new Headers({ 'location': path })
    });
  }

  static badRequest(message: string): Response {
    return Response.fromHttpError(new HttpError.BadRequest(message));
  }

  static notFound(path: string): Response {
    return Response.fromHttpError(new HttpError.NotFound(path));
  }

  static serverFailure(failure: Failure): Response {
    return Response.fromHttpError(new HttpError.InternalServerError(failure.message, failure.cause));
  }
}
